src/home_robot_sim/home_robot_sim/env/habitat_ovmm_env/habitat_ovmm_env.py

```diff
@@ -298,20 +299,25 @@ def _preprocess_action(
             cont_action = np.concatenate(
                 [arm_action, grip_action, base_vel, [-1, -1, rearrange_stop[0], -1]]
             )
-        elif type(action) == ContinuousNavigationAction:
+        elif type(action) in [ContinuousFullBodyAction, ContinuousNavigationAction]:
             grip_action = -1
             # Keep holding in case holding an object
             if habitat_obs["is_holding"][0] == 1:
                 grip_action = 1
             waypoint_x, waypoint_y, turn = 0, 0, 0
-            if action.xyt[0] != 0:
-                waypoint_x = np.clip(action.xyt[0] / self.max_forward, -1, 1)
-            elif action.xyt[1] != 0:
-                waypoint_y = np.clip(action.xyt[1] / self.max_forward, -1, 1)
-            elif action.xyt[2] != 0:
-                turn = np.clip(action.xyt[2] / self.max_turn, -1, 1)
+            if action.xyt is not None:
```

Comment:
Some more comments describing what's happening here would be useful; there's a lot of logic